processor/spanmetricsprocessor/processor.go

```diff
@@ -237,6 +237,15 @@ func (p *processorImp) buildMetrics() *pdata.Metrics {
 	p.lock.RLock()
 	p.collectCallMetrics(ilm)
 	p.collectLatencyMetrics(ilm)
+
+	// If delta metrics, flush accumulated data
+	if p.config.GetAggregationTemporality() == pdata.MetricAggregationTemporalityDelta {
+		p.callSum = make(map[metricKey]int64)
+		p.latencyCount = make(map[metricKey]uint64)
+		p.latencySum = make(map[metricKey]float64)
+		p.latencyBucketCounts = make(map[metricKey][]uint64)
+	}
```

Comment:
Can I ask why we had opted for overriding the map values? Was there a performance improvement compare to deleting all the entries in the map?

Reply:
Mainly so I don't have to loop over every map, and it makes the code simpler. If you feel strongly either way I'm happy for my mind to be changed.